### PR TITLE
Deadlock fixes

### PR DESCRIPTION
**Pull request by crimsonred**

### CODE REVIEW

File: examples/cli/cli_demo.go [hunk 1]
```diff
@@ -2196,11 +2198,25 @@ func publishRequest(args []string) {
 		"m2": "n2",
 	}
 
+	payload := map[string]interface{}{
```
Comment (davidnub) [inline examples/cli/cli_demo.go]: This doesn't look related to deadlocks. :)

Comment (crimsonred) [reply examples/cli/cli_demo.go]: This was for the APNS test, I used the example and didn't remove it :(
